src/components/Modals/Send/views/Details.tsx

```diff
@@ -83,7 +83,8 @@ export const Details = () => {
   useEffect(() => {
     // Initial setting of cryptoAmount in case of a QR-code set amount
     handleInputChange(cryptoAmount ?? '0')
-  }, [cryptoAmount, handleInputChange])
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [handleInputChange])
```

Comment:
The guts of the fiat amount reactivity issue - we *don't* want to be reactive on `cryptoAmount` here. This should really be ran once when `<Details />` mounts, which will fire `handleInputAmount` **once** with the amount from the QR code.
This should *not* fire again on input change, else it will produce zero values in case the current input is a fiat amount.

Reply:
It doesn't, this previously worked for crypto already in the reverted PR, but not for fiat and will now only run on `handleInputChange` referential instability.

The reason it was previously borked on fiat amounts is because it was running on `cryptoAmount` / `fiatAmount` referential instability (a.ka on keyboard input), but running `handleInputChange(cryptoAmount)` which is obviously incorrect - it should be run by both `<TokenRow />`s respective handlers, with crypto amount if within the context of a crypto amount, and fiat amount in the context of a fiat amount

Ran another paranoia test here:

<img width="517" alt="image" src="https://user-images.githubusercontent.com/17035424/236014893-051bd307-3a2e-4850-9859-a491fd3ec608.png">
<img width="533" alt="image" src="https://user-images.githubusercontent.com/17035424/236015032-becc3d88-d1f1-422a-97b9-ac7c91bd1bda.png">
<img width="495" alt="image" src="https://user-images.githubusercontent.com/17035424/236015056-cb4b2ff3-c87d-494d-a998-96f7039a1baf.png">
<img width="506" alt="image" src="https://user-images.githubusercontent.com/17035424/236015093-8faa0ea3-0e1b-48a3-b44d-33dae23da091.png">
<img width="495" alt="image" src="https://user-images.githubusercontent.com/17035424/236015494-b0651dc5-79be-46fa-ba02-fd4401053c3e.png">

While trying to break it, found out that toggling to fiat works as tested, but back to crypto brings borked amounts. The reason for that is `handleInputChange` is reactive in itself since it's part of a hook. So even this reactivity is too much here, we want no deps at all - 4b58fcfc988686ff5c1d5f11d1837f4efbef2f5c

Reply:
Actually those forms are proper 🥖 - this will bork crypto balance validation on QR code button sends since we set the form values outside the component - `trigger()` didn't help here - https://github.com/shapeshift/web/pull/4406/commits/29662d7a1fbfd45d913c4528a4220002c042421e